var/spack/repos/builtin/packages/py-fenics-dolfinx/package.py

```diff
@@ -58,7 +58,8 @@ class PyFenicsDolfinx(PythonPackage):
     depends_on("py-petsc4py", type=("build", "run"))
     depends_on("py-cffi", type=("build", "run"))
 
-    depends_on("py-nanobind@1.8:", when="@0.8:", type="build")
+    depends_on("py-nanobind@2:", when="@0.9:", type="build")
```

Comment:
Do you want to be explicit about `0.9` here now (versus maybe `@main`) since `0.9.0` isn't out yet and there are additional constraints on the `py-fenicx-*` packages that will need to be updated when it does come out.

Reply:
I think we're happy with this - the way it is we won't have to change this line once v0.9 is released.

Reply:
That is true.  You'll just need to add lines for `py-fenics-*` dependencies.